site/public/js/drag-and-drop.js

```diff
@@ -1297,6 +1298,116 @@ function handleUploadCourseMaterials(csrf_token, expand_zip, hide_from_students,
+    formData.append('close_date', closeDate);
+    $.ajax({
+        url: buildUrl(['banner', 'delete']),
+        data: formData,
+        processData: false,
+        contentType: false,
+        type: 'POST',
+        success: function(data) {
+            try {
+                const jsondata = JSON.parse(data);
+
+                if (jsondata['status'] === 'success') {
+                    window.location.href = buildUrl(['banner']);
+                }
+                else {
+                    alert(jsondata['message']);
+                }
+            }
+            catch (e) {
+                alert('Error parsing response from server. Please copy the contents of your Javascript Console and ' +
+                    'send it to an administrator, as well as what you were doing and what files you were uploading. - [handleUploadCourseMaterials]');
+                console.log(data);
+            }
+        },
+        error: function() {
+            window.location.href = buildUrl(['banner']);
+        },
+    });
+}
+
+
+/**
+ * @param csrf_token
+ */
+
+function handleUploadBanner(csrf_token, closeTime, releaseTime, extraName) {
+    const formData = new FormData();
+    formData.append('csrf_token', csrf_token);
+    formData.append('close_time', closeTime);
+    formData.append('release_time', releaseTime);
+    formData.append('extra_name', extraName);
+    for (let i = 0; i < file_array.length; i++) {
+        for (let j = 0; j < file_array[i].length; j++) {
+            if (file_array[i][j].name.indexOf("'") !== -1 ||
+              file_array[i][j].name.indexOf('"') !== -1) {
+                alert(`ERROR! You may not use quotes in your filename: ${file_array[i][j].name}`);
+                return;
+            }
+            else if (file_array[i][j].name.indexOf('\\') !== -1 ||
+              file_array[i][j].name.indexOf('/') !== -1) {
+                alert(`ERROR! You may not use a slash in your filename: ${file_array[i][j].name}`);
+                return;
+            }
+            else if (file_array[i][j].name.indexOf('<') !== -1 ||
+              file_array[i][j].name.indexOf('>') !== -1) {
+                alert(`ERROR! You may not use angle brackets in your filename: ${file_array[i][j].name}`);
+                return;
+            }
+            const k = fileExists(`/${file_array[i][j].name}`, 1);
+            // Check conflict here
+            if ( k[0] === 1 ) {
+                if (!confirm(`Note: ${file_array[i][j].name} already exists. Do you want to replace it?`)) {
+                    continue;
+                }
+            }
+            formData.append(`files${i + 1}[]`, file_array[i][j], file_array[i][j].name);
+        }
+    }
+    $.ajax({
+        url: buildUrl(['banner', 'upload']),
+        data: formData,
+        processData: false,
+        contentType: false,
+        type: 'POST',
+        success: function(data) {
+            try {
+                const jsondata = JSON.parse(data);
```

Comment:
what is the reason to parse the data as json here? 
This formdata is object, you can directly do data.status === 'success' instead of parsing it json, then matching it with the keyword.